src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int INIT_SIZE = 10;
+    private static final double ZOOM_SIZE = 1.5;
+    private Object[] elementData = new Object[INIT_SIZE];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (elementData.length == size) {
+            resize();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexForAdd(index);
+        if (elementData.length == size) {
+            resize();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int numNew = list.size();
+        for (int i = 0; i < numNew; i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = (T) elementData[index];
+        fastRemove(elementData, index);
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/200846632-28869b57-8a28-48c6-b889-ad45e3e0d851.png)

Reply:
keep it simple